reference/commands/custom_commands.rst

```diff
@@ -0,0 +1,139 @@
+
+The call to those commands change a little bit: :command:`conan [topic_name]:your_command_name`. Following the previous example:
+
+.. code-block:: text
+
+    $ conan greet:hello
+    $ conan greet:bye
+
+
+Decorators
+-----------
+
+conan_command(group=None, formatters=None)
++++++++++++++++++++++++++++++++++++++++++++
+
+Main decorator to declare a function as a new Conan command. Where the parameters are:
+
+* ``group`` is the name of the group of commands declared under the same name.
+  This grouping will appear executing the :command:`conan -h` command.
+* ``formatters`` is a dict-like Python object where the ``key`` is the formatter name and the value is the
+  function instance where will be processed the information returned by the command one.
+
+
+.. code-block:: python
+    :caption: cmd_hello.py
+
+    import json
+
+    from conan.api.conan_api import ConanAPIV2
+    from conans.cli.command import conan_command
+    from conans.cli.output import ConanOutput
+
+
+    def output_json(msg):
+        return json.dumps({"greet": msg})
+
+
+    @conan_command(group="custom commands", formatters={"json": output_json})
+    def hello(conan_api: ConanAPIV2, parser, *args):
+        """
+        Simple command to print "Hello World!" line
+        """
+        msg = "Hello World!"
+        ConanOutput().info(msg)
+        return msg
+
+
+.. important::
+
+    The function decorated by ``@conan_command(....)`` must have the same name as the suffix used by the Python file.
+    For instance, the previous example, the file name is ``cmd_hello.py``, and the command function decorated is ``def hello(....)``.
+
+
+conan_subcommand(formatters=None)
+++++++++++++++++++++++++++++++++++++
+
+Similar to ``conan_command``, but this one is declaring a sub-command of an existing custom command. For instance:
+
+.. code-block:: python
+    :caption: cmd_hello.py
+
+    from conan.api.conan_api import ConanAPIV2
+    from conans.cli.command import conan_command, conan_subcommand
+    from conans.cli.output import ConanOutput
+
+
+    @conan_subcommand()
+    def hello_moon(conan_api, parser, subparser, *args):
+        """
+        Sub-command of "hello" that prints "Hello Moon!" line
+        """
+        ConanOutput().info("Hello Moon!")
+
+
+    @conan_command(group="Custom commands")
+    def hello(conan_api: ConanAPIV2, parser, *args):
+        """
+        Simple command "hello"
+        """
+
+The command call looks like :command:`conan hello moon`.
+
+
+Command function arguments
+----------------------------
+
+These are the passed arguments to any custom command and its sub-commands functions:
+
+.. code-block:: python
+    :caption: cmd_command.py
+
+    from conans.cli.command import conan_command, conan_subcommand
+
+    @conan_subcommand()
+    def command_subcommand(conan_api, parser, subparser, *args):
+        pass
+
+    @conan_command(group="Custom commands")
+    def command(conan_api, parser, *args):
+        pass
```

Comment:
again, empty functions. it would be better to illustrate with something real, otherwise it's a bit unclear for reader what should be done in that functions.

Reply:
okay, I think that deserves some comment or explanation in doc.